src/main/java/com/aws/greengrass/mqttclient/MqttClient.java

```diff
@@ -559,8 +572,8 @@ public boolean connected() {
     @Override
     public synchronized void close() {
         // Shut down spooler and then no more message will be published
-        if (spoolingFuture.get() != null) {
-            spoolingFuture.get().cancel(true);
+        if (isSpooling.get()) {
+            executorService.shutdownNow();
```

Comment:
Absolutely do not do this. You don't own the executor service, you may not shut it down.

Reply:
Revert this back to how it was, that was correct.